analysis/druidbalance/src/modules/features/BalanceOfAllThingsOpener.tsx

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SPELLS from 'common/SPELLS';
+import { SpellLink } from 'interface';
+import { t } from '@lingui/macro';
+import Events, {
+  ApplyBuffEvent,
+  CastEvent,
+  DamageEvent,
+  RemoveBuffEvent,
+} from 'parser/core/Events';
+import GlobalCooldown from 'parser/shared/modules/GlobalCooldown';
+
+class BalanceOfAllThingsOpener extends Analyzer {
+  /*
+    This module is used to determine wether there were three starsurge cast used to open a Balance of All Things window when in a single target situation.
+    There are several special situations that need to be filtered out in order to avoid unnecessary warnings:
+    1. When currently in a multitarget situation one might not want to spend his AP on starsurges, in order to keep up starfall
+    2. When at the start of the fight you do not have enough AP to open up with 3 starsurges
+    3. When using Convoke the Spirits you do not open with starsurges
+    */
+  static dependencies = {
+    globalCooldown: GlobalCooldown,
+  };
+
+  constructor(options: Options) {
+    super(options);
+
+    const active = this.selectedCombatant.hasLegendaryByBonusID(
+      SPELLS.BALANCE_OF_ALL_THINGS_SOLAR.bonusID,
+    );
+    if (!active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BALANCE_OF_ALL_THINGS_SOLAR),
+      this.onApplyBuff,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.BALANCE_OF_ALL_THINGS_LUNAR),
+      this.onApplyBuff,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BALANCE_OF_ALL_THINGS_SOLAR),
+      this.onRemoveBuff,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.BALANCE_OF_ALL_THINGS_LUNAR),
+      this.onRemoveBuff,
+    );
```

Comment:
For future reference, spell can accept an array of abilities so this could be condensed into:
```ts
Events.removebuff.by(SELECTED_PLAYER).spell([SPELLS.BALANCE_OF_ALL_THINGS_SOLAR, SPELLS.BALANCE_OF_ALL_THINGS_LUNAR])
```